### PR TITLE
UPSTREAM: <carry>: drop in 4.17 once all clusters have been through 4.15

### PR DESCRIPTION
**Pull request by deads2k**
straight-forward fix?

/assign @p0lyn0mial

### CODE REVIEW

File: pkg/apis/core/validation/validation.go [hunk 1]
```diff
@@ -6407,7 +6407,9 @@ func ValidateSecret(secret *core.Secret) field.ErrorList {
 func ValidateSecretUpdate(newSecret, oldSecret *core.Secret) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&newSecret.ObjectMeta, &oldSecret.ObjectMeta, field.NewPath("metadata"))
 
-	allErrs = append(allErrs, ValidateImmutableField(newSecret.Type, oldSecret.Type, field.NewPath("type"))...)
+	if oldSecret.Namespace != "openshift-kube-apiserver-operator" && oldSecret.Namespace != "openshift-kube-apiserver" {
```
Comment (p0lyn0mial) [inline pkg/apis/core/validation/validation.go:6410]: it looks like we need to patch https://github.com/openshift/library-go/blob/2c99390d34b0fd086758f66035a768191a67a1e9/pkg/operator/resource/resourceapply/core.go#L429 as well - to stop deletion.

Comment (deads2k) [reply pkg/apis/core/validation/validation.go:6410]: in https://github.com/openshift/library-go/pull/1703

**Comment by deads2k** [comment]
I think the combination of the two
1. ensures we never delete the secret, thus fixing the failure mode.
2. ensures we end up with closer to identical upgraded vs new clusters, thus fixing the testing gap.

Any other improvement is also welcome, but these two are required since they are very easy to reason about we 're assured the deletion won't happen and that our testing gap is closer to getting fixed.

File: pkg/apis/core/validation/validation.go (hunk 1 above)
Comment (p0lyn0mial) [reply pkg/apis/core/validation/validation.go:6410]: @deads2k why do you want to also relax the validation for `openshift-kube-apiserver` ?

**Comment by p0lyn0mial** [comment]
/close

in favour of https://github.com/openshift/kubernetes/pull/1924 and https://github.com/openshift/kubernetes/pull/1929

**Pull request closed without merging**